crates/voicevox_core/src/engine/kana_parser.rs

```diff
@@ -0,0 +1,280 @@
+                        consonant_length: None,
+                        vowel: "pau".to_string(),
+                        vowel_length: 0.0,
+                        pitch: 0.0,
+                    });
+                }
+                accent_phrase.is_interrogative = is_interrogative;
+                accent_phrase
+            };
+            parsed_result.push(accent_phrase);
+            phrase.clear();
+        } else {
+            phrase.push(letter);
+        }
+    }
+    Ok(parsed_result)
+}
+
+#[allow(dead_code)] // TODO: remove this feature
+fn create_kana(accent_phrases: &[AccentPhraseModel]) -> String {
+    let mut text = String::new();
+    for phrase in accent_phrases {
+        let moras = &phrase.moras;
+        for (index, mora) in moras.iter().enumerate() {
+            if ["A", "E", "I", "O", "U"].contains(&mora.vowel.as_ref()) {
+                text.push(UNVOICE_SYMBOL);
+            }
+            text.push_str(&mora.text);
+            if index + 1 == phrase.accent {
+                text.push(ACCENT_SYMBOL);
+            }
+        }
+        if phrase.is_interrogative {
+            text.push(WIDE_INTERROGATION_MARK);
+        }
+        text.push(if phrase.pause_mora.is_some() {
+            PAUSE_DELIMITER
+        } else {
+            NOPAUSE_DELIMITER
+        });
+    }
+    text.pop(); // remove last delimiter
+    text
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::engine::mora_list::MORA_LIST_MINIMUM;
+    use pretty_assertions::assert_eq;
+    use rstest::rstest;
+
+    #[rstest]
+    #[case(Some("da"), "ダ")]
+    #[case(Some("N"), "ン")]
+    #[case(Some("cl"), "ッ")]
+    #[case(Some("sho"), "ショ")]
+    #[case(Some("u"), "ウ")]
+    #[case(Some("gA"), "_ガ")]
+    #[case(Some("byO"), "_ビョ")]
+    #[case(Some("O"), "_オ")]
+    #[case(None, "fail")]
+    fn test_text2mora_with_unvoice(#[case] mora: Option<&str>, #[case] text: &str) {
+        let text2mora = &TEXT2MORA_WITH_UNVOICE;
+        assert_eq!(text2mora.len(), MORA_LIST_MINIMUM.len() * 2 - 2); // added twice except ン and ッ
+        let res = text2mora.get(text);
+        assert_eq!(mora.is_some(), res.is_some());
+        if let Some(res) = res {
+            let mut m = String::new();
+            if let Some(ref c) = res.consonant {
+                m.push_str(c);
+            }
+            m.push_str(&res.vowel);
+            assert_eq!(m, mora.unwrap());
+        }
+    }
+
+    #[test]
+    fn test_text_to_accent_phrase() {
+        let text_ok = ["ア_シタ'ワ", "ユウヒガ'", "_キ'レイ"];
+        let text_err = [
+            "アクセントナシ",
+            "アクセ'ント'タクサン'",
+            "'アクセントハジマリ",
+            "不明な'文字",
+        ];
+        for text in text_ok {
+            let result = text_to_accent_phrase(text);
+            assert!(result.is_ok(), "{:?}", result);
+        }
+        for text in text_err {
+            let result = text_to_accent_phrase(text);
+            assert!(result.is_err(), "{:?}", result);
+        }
+    }
```

Comment:
このテスト関数も rstest で書くと、他のテストと形式が同じになるので後で読みやすくなると思います。

`KanaParseError` に `PartialEq` を実装して `result` と直接比較できるようにすると `expected` として `KanaParseResult` を渡せるのでわかりやすく書けそうですが、エラーメッセージを比較したい訳でもないので、書くとしたら以下のように `expected` として `bool` 値を渡す書き方が良いかも、と思いました（前に `assert!` の書き方についてコメントした手前、`assert_eq!` を書くという違う主張になってしまいすみません……！）。

```suggestion
    #[rstest]
    #[case("ア_シタ'ワ", true)]
    #[case("ユウヒガ'", true)]
    #[case("_キ'レイ", true)]
    #[case("アクセントナシ", false)]
    #[case("アクセ'ント'タクサン'", false)]
    #[case("'アクセントハジマリ", false)]
    #[case("不明な'文字", false)]
    fn test_text_to_accent_phrase(#[case] text: &str, #[case] result_is_ok_expected: bool) {
        let result = text_to_accent_phrase(text);
        assert_eq!(result.is_ok(), result_is_ok_expected, "{:?}", result);
    }
```